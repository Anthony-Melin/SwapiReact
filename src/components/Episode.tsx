import LabelValueFactory from './__components__/LabelValueFactory.tsx'

export default LabelValueFactory('episode', 'Episode')
