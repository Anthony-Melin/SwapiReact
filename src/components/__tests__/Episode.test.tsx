import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Episode from '../Episode.tsx'

describe('Episode', () => {
    beforeEach(() => {
        render(<Episode value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('episode')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('episode')).toHaveTextContent(
            'Episode: VALUE'
        )
    })
})
