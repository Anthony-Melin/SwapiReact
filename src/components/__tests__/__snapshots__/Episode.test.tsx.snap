// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Episode > should match snapshot 1`] = `
<p
  data-testid="episode"
>
  Episode
  : 
  VALUE
</p>
`;
